docker/entrypoint.sh

```diff
@@ -9,4 +11,23 @@ if [ -n "$METADATA_AGENT_URL" ]; then
     /etc/google-fluentd/google-fluentd.conf
 fi
 
-/usr/sbin/google-fluentd "$@"
+trap 'kill $pid' TERM INT
+
+# first arg is a flag.
+if [ "${1:0:1}" = '-' ]; then
+  /usr/sbin/google-fluentd "$@" &
+else
+  exec "$@" &
+fi
+
+pid=$!
+wait $pid
+exit_code="$?"
+wait $pid
+
+# Ignore 143, as that is the code when SIGINT kills the process.
```

Comment:
I think you meant SIGTERM. SIGINT is 130.

Reply:
SIGINT is the default kill signal sent by `docker stop`, that's the code I'm referring to, probably should make that clearer.

Reply:
Yeah, maybe rephrase the comment a bit to make that clear.